### ISSUE TITLE
[backend] Works tracking fail to complete at "update processed time"

### ISSUE THREAD

**Issue by richard-julien** [LuatixHQ/opencti#5762]
## Description

Auto complete works at processed time update fail because of type enforcing in the search engine

`illegal_argument_exception: Integer value passed as String`

Data in redis are stored in string so we need to convert this figures to int before using it.

Code can be refactored to only use one function (currently one is ok and the other not)

**Comment by MaxwellDPS** [comment]
Still seeing this on 5.12.33 FYSA 

```json
{
    "category": "APP",
    "error": {
        "_error": {},
        "_showLocations": false,
        "_showPath": false,
        "data": {
            "_error": {},
            "_showLocations": false,
            "_showPath": false,
            "data": {
                "body": {
                    "script": {
                        "lang": "painless",
                        "params": {
                            "completed_number": "159333",
                            "completed_time": "2024-03-15T20:11:18.222Z"
                        },
                        "source": "ctx._source['status'] = \"complete\";\n                  ctx._source['completed_time'] = params.completed_time;\n                  ctx._source['completed_number'] = params.completed_number;"
                    }
                },
                "cause": {
                    "meta": {
                        "body": {
                            "error": {
                                "caused_by": {
                                    "reason": "Integer value passed as String",
                                    "type": "illegal_argument_exception"
                                },
                                "reason": "[1:631] failed to parse field [completed_number] of type [integer] in document with id 'work_<REMOVED>_2024-03-13T20:46:35.244Z'. Preview of field's value: '159333'",
                                "root_cause": {
                                    "0": {
                                        "reason": "[1:631] failed to parse field [completed_number] of type [integer] in document with id 'work_<REMOVED>_2024-03-13T20:46:35.244Z'. Preview of field's value: '159333'",
                                        "type": "document_parsing_exception"
                                    }
                                },
                                "type": "document_parsing_exception"
                            },
                            "status": 400
                        },
                        "headers": {
                            "content-length": "611",
                            "content-type": "application/vnd.elasticsearch+json;compatible-with=8",
                            "x-elastic-product": "Elasticsearch"
                        },
                        "meta": {
                            "aborted": false,
                            "attempts": 0,
                            "connection": {
                                "_openRequests": 0,
                                "deadCount": 0,
                                "headers": {
                                    "authorization": "[redacted]"
                                },
                                "id": "<REMOVED>",
                                "pool": {
                                    "_events": {},
                                    "_eventsCount": 0
                                },
                                "resurrectTimeout": 0,
                                "timeout": 30000,
                                "tls": {},
                                "url": "<REMOVED>",
                                "weight": 1000
                            },
                            "context": null,
                            "name": "elasticsearch-js",
                            "request": {
                                "id": 83283,
                                "options": {},
                                "params": {
                                    "body": "{\"script\":{\"source\":\"ctx._source['status'] = \\\"complete\\\";\\n                  ctx._source['completed_time'] = params.completed_time;\\n                  ctx._source['completed_number'] = params.completed_number;\",\"lang\":\"painless\",\"params\":{\"completed_time\":\"2024-03-15T20:11:18.222Z\",\"completed_number\":\"159333\"}}}",
                                    "headers": {
                                        "accept": "application/vnd.elasticsearch+json; compatible-with=8",
                                        "content-length": "314",
                                        "content-type": "application/vnd.elasticsearch+json; compatible-with=8",
                                        "user-agent": "elasticsearch-js/8.11.0 (linux 5.14.0-319.el9.x86_64-x64; Node.js 20.11.1; Transport 8.4.0)",
                                        "x-elastic-client-meta": "es=8.11.0,js=20.11.1,t=8.4.0,hc=20.11.1"
                                    },
                                    "method": "POST",
                                    "path": "/opencti_history-000002/_update/work_<REMOVED>_2024-03-13T20%3A46%3A35.244Z",
                                    "querystring": "retry_on_conflict=5&timeout=5m&refresh=true"
                                }
                            }
                        },
                        "statusCode": 400,
                        "warnings": null
                    },
                    "name": "ResponseError",
                    "options": {
                        "redaction": {
                            "additionalKeys": [],
                            "type": "replace"
                        }
                    }
                },
                "documentId": "work_<REMOVED>_2024-03-13T20:46:35.244Z",
                "genre": "TECHNICAL",
                "http_status": 500
            },
            "internalData": {},
            "name": "DATABASE_ERROR",
            "time_thrown": "2024-03-15T20:11:18.251Z"
        },
        "level": "info",
        "message": "[OPENCTI-MODULE] Connector manager error processing work closing",
        "timestamp": "2024-03-15T20:11:18.252Z",
        "version": "5.12.33"
    }
}
```

**Comment by richard-julien** [comment]
Hi @MaxwellDPS , correct. This one was fixed in 6.0
